### ISSUE TITLE
multi column addIndex with limit specifier

### ISSUE THREAD

**Issue by glensc** [cakephp/phinx#1080]
i'm unable to produce index specifying length of individual fields. this is mysql table.

```mysql
CREATE TABLE `email_account` (
  `ema_folder` varchar(255) DEFAULT NULL,
  `ema_hostname` varchar(255) NOT NULL DEFAULT '',
  `ema_username` varchar(64) NOT NULL DEFAULT '',
  UNIQUE KEY `ema_username` (`ema_username`,`ema_hostname`(100),`ema_folder`(100))
);
```

i tried:

```php
->addIndex(['ema_username', 'ema_hostname', 'ema_folder'], ['unique' => true, 'limit' => 100])
```

however that tries to execute:

```mysql
UNIQUE KEY (`ema_username`,`ema_hostname`,`ema_folder`(100)),  KEY (`ema_prj_id`))
```

which in my case fails with mysql engine:

```
  SQLSTATE[42000]: Syntax error or access violation: 1071 Specified key was too long; max key length is 1000 bytes  
```

**Comment by glensc** [comment]
looks like the current `Index` structure does not allow per-column limit, so it's a bit bigger work to handle this.

https://github.com/robmorgan/phinx/blob/v0.8.0/src/Phinx/Db/Table/Index.php#L66

**Comment by dereuromark** [comment]
Are you able to provide a patch with your suggested changes as PR?

**Comment by BLaDZer** [comment]
I think this should be closed cuz it fixed long time ago in v0.10.0
